### ISSUE TITLE
Error on getting all boards

### ISSUE THREAD

**Issue by jeremieca** [jira-node/node-jira-client#225]
Hello,

I have an error on getAllBoards() method.

This is my code : 

```js
        jira.getAllBoards()
            .then((jiraOutput) => {
                ......
            })
            .catch(reject);
```

The error is : 

```
.../node_modules/bluebird/js/release/debuggability.js:868 Unhandled rejection StatusCodeError: 400 - {"errorMessages":["No project could be found with key ''."],"errors":{}}
```

I found the problem in node-jira-client, it's on getAllBoards method. When there is no projectKeyOrId in argument, param is passed in query string as empty. So Jira try to filter on an empty project name : 

```js
return this.doRequest(this.makeRequestHeader(this.makeAgileUri({
        pathname: '/board',
        query: {
          startAt: startAt,
          maxResults: maxResults,
          type: type,
          name: name,
          projectKeyOrId: projectKeyOrId
        }
})));
```

In the case where I really want to get all board (whithout filtering on project key), I need to remove projectKeyOrId key in query object. And it works.

**Comment by LeonAlvarez** [comment]
Pull request to fix this issue #227

**Comment by pioug** [comment]
Fixed in https://github.com/jira-node/node-jira-client/pull/247
